src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js

```diff
@@ -74,19 +74,21 @@ export class CreateBaselineModal extends Component {
     };
 
     async componentDidMount() {
-        window.entityListener = addNewListener({
-            actionType: 'SELECT_ENTITY',
-            callback: () => {
-                this.props.createBaselineModalOpened ? this.deselectHistoricalProfiles() : null;
-            }
-        });
+        if (this.props.middlewareListener) {
```

Comment:
This isn't nice... but it seems to be difficult to do without having to do a lot of changes.